packages/pwa-devdocs/src/venia-pwa-concept/design/global-style/index.md

```diff
@@ -0,0 +1,151 @@
+---
+title: Global style guide
+---
+
+{: .bs-callout .bs-callout-info}
+**Note:**
+This design specification is still in development.
+If have any feedback or would like to join the PWA conversation, please join our [Slack][] channel.
+
+This topic provides the global style rules and values used in the Venia theme.
+Use this guide to style components that will fit within the Venia theme.
+
+## Colors
+
+{: .bs-callout .bs-callout-tip}
+***Tip:** Color values are expressed in RGB format because it translates easier to `rgba()` values than HEX format.*
+
+The Venia theme uses the following color classes in its design:
+
+| CSS class         | RGB value                     | Usage            |
+| ----------------- | ----------------------------- | ---------------- |
+| `black`           | `rgb(0, 0, 0)`                |                  |
+| `venia-border`    | `rgb(224, 224, 224)`          | Border color     |
+| `venia-grey`      | `rgb(246, 246, 246)`          |                  |
+| `venia-teal`      | `rgb(0, 134, 139)`            | Buttons          |
+| `venia-text`      | `rgb(33, 33, 33)`             | General text     |
+| `venia-text-alt`  | `rgb(117, 117, 117)`          | Alternative text |
+| `venia-text-hint` | `rgb(158, 158, 158)`          | Hint text        |
+| `mask.root`       | `rgb(0, 0, 0)` at 50% opacity | Mask overlay     |
```

Comment:
`mask.root` is not a global style, nor is it a classname that people can apply, so we should consider removing it from this list for now. In the future, though, we could capture part of this style as a custom property and update this document at that time.

Reply:
Text color variables that are shades of pure gray should probably use `rgba(0, 0, 0, 0.5)`. Otherwise the middle grays tend to wash out against a non-white background as it darkens, but with a black-based `rgba()` they'll get darker and retain relative contrast.

Reply:
As you might expect, these grays are actually pre-blended tints of black. The real data here would be opacity.

opacity | blended on white
--- | ---
`0.87` | `rgb(33, 33, 33)`
`0.54` | `rgb(117, 117, 117)`
`0.38` | `rgb(158, 158, 158)`

We don't use `rgba()` for text because of how text rendering works in browsers—text appears thinner when it's not fully opaque—but that does introduce the problem you pointed out, which is that the text washes out against non-white backgrounds. Ordinarily, the solution would be to set the text color when setting the background color:

```stylus
.foo
    background-color: rgb(236, 232, 245)
    color: blend(rgba(black, 0.87), @background-color)
```

And that's normally what I've done, but without a preprocessor, I'm not sure how to accomplish it.

Reply:
Thinning is a good point. I haven't worried about this as much since the effect isn't as pronounced with text > 16px and @2x screens becoming the norm. Made a CodePen to test, and it does look a bit bad on @1x screens:

https://codepen.io/brendanfalkowski/pen/djwNpy

To address wash-out, in design systems I've built there's usually a concept of "tones" as a token. They're not specific to components, but they makeup the majority of component coloration. The component applies a tone rather than defining hex/rgb colors.

- color-background
- color-border
- color-text-base
- color-text-subtle

![screen shot 2018-08-09 at 12 58 49 pm](https://user-images.githubusercontent.com/214924/43922781-9646ed26-9bd4-11e8-8c7d-713fd436bf30.png)

That type of organization helps avoid the mishaps of consistently changing text styles when the background changes.

Doesn't help doing color math without a preprocessor though.